src/commands/generateFunctions.ts

```diff
@@ -0,0 +1,241 @@
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { OpenDialogOptions, ProgressLocation, Uri, window, workspace } from "vscode";
+import { DialogResponses } from 'vscode-azureextensionui';
+import XRegExp = require('xregexp');
+import { ApiTreeItem } from "../explorer/ApiTreeItem";
+import { OpenApiEditor } from "../explorer/editors/openApi/OpenApiEditor";
+import { ext } from "../extensionVariables";
+import { localize } from "../localize";
+import { cpUtils } from "../utils/cpUtils";
+import { openUrl } from '../utils/openUrl';
+
+const formattingCharacter: string = '\\p{Cf}';
+const connectingCharacter: string = '\\p{Pc}';
+const decimalDigitCharacter: string = '\\p{Nd}';
+const combiningCharacter: string = '\\p{Mn}|\\p{Mc}';
+const letterCharacter: string = '\\p{Lu}|\\p{Ll}|\\p{Lt}|\\p{Lm}|\\p{Lo}|\\p{Nl}';
+const identifierPartCharacter: string = `${letterCharacter}|${decimalDigitCharacter}|${connectingCharacter}|${combiningCharacter}|${formattingCharacter}`;
+const identifierStartCharacter: string = `(${letterCharacter}|_)`;
+const identifierOrKeyword: string = `${identifierStartCharacter}(${identifierPartCharacter})*`;
+// tslint:disable-next-line: no-unsafe-any
+const identifierRegex: RegExp = XRegExp(`^${identifierOrKeyword}$`);
+// Keywords: https://github.com/dotnet/csharplang/blob/master/spec/lexical-structure.md#keywords
+const keywords: string[] = ['abstract', 'as', 'base', 'bool', 'break', 'byte', 'case', 'catch', 'char', 'checked', 'class', 'const', 'continue', 'decimal', 'default', 'delegate', 'do', 'double', 'else', 'enum', 'event', 'explicit', 'extern', 'false', 'finally', 'fixed', 'float', 'for', 'foreach', 'goto', 'if', 'implicit', 'in', 'int', 'interface', 'internal', 'is', 'lock', 'long', 'namespace', 'new', 'null', 'object', 'operator', 'out', 'override', 'params', 'private', 'protected', 'public', 'readonly', 'ref', 'return', 'sbyte', 'sealed', 'short', 'sizeof', 'stackalloc', 'static', 'string', 'struct', 'switch', 'this', 'throw', 'true', 'try', 'typeof', 'uint', 'ulong', 'unchecked', 'unsafe', 'ushort', 'using', 'virtual', 'void', 'volatile', 'while'];
+
+// tslint:disable: no-non-null-assertion
+export async function generateFunctions(node?: ApiTreeItem): Promise<void> {
+    if (!node) {
+        node = <ApiTreeItem>await ext.tree.showTreeItemPicker(ApiTreeItem.contextValue);
+    }
+
+    ext.outputChannel.show();
+
+    const openApiEditor: OpenApiEditor = new OpenApiEditor();
+    const openAPIDocString = await openApiEditor.getData(node);
+
+    if (!await validateAutorestInstalled()) {
+        return;
```

Comment:
Add a log-line for the output to specify why this failed.

Reply:
we already provide the error message card, it will be duplicated

Reply:
Ah got it - [R180](https://github.com/microsoft/vscode-apimanagement/pull/121/files#diff-0a195892fd8b16b28a2e378bddd686a8R180). I assumed `validateAutorestInstalled` only validated - based on the name.

Reply:
Similar to how you did `checkEnvironmentInstalled`.